src/main/org/codehaus/groovy/ast/tools/GeneralUtils.java

```diff
@@ -219,10 +219,38 @@ public static ClosureExpression closureX(Statement code) {
         return result;
     }
 
+    /**
+     * Build a binary expression that compares two values
+     * @param lhv expression for the value to compare from
+     * @param rhv expression for the value value to compare to
+     * @return the expression comparing two values
+     */
     public static BinaryExpression cmpX(Expression lhv, Expression rhv) {
         return new BinaryExpression(lhv, CMP, rhv);
     }
 
+    /**
+     * Build a binary expression that compares two values
+     * @param lhv expression for the value to compare from
+     * @param rhv expression for the value value to compare to
+     * @param reversed whether to use natural ordering or reversed natural ordering
+     * @return the expression comparing two values
+     */
+    public static BinaryExpression cmpX(Expression lhv, Expression rhv, boolean reversed) {
+        return (reversed) ? cmpXReversed(lhv, rhv) : cmpX(lhv, rhv);
```

Comment:
What about `return (reversed) ? cmpX(rhv, lhv) : cmpX(lhv, rhv);`?  Could eliminate the need for the new `cmpXReversed` method.